metrics/config_test.go

```diff
@@ -30,27 +27,14 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 
-	"go.opencensus.io/stats"
-	"go.opencensus.io/stats/view"
-
 	. "knative.dev/pkg/logging/testing"
-	"knative.dev/pkg/metrics/metricstest"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// TODO UTs should move to eventing and serving, as appropriate.
-// 	See https://github.com/knative/pkg/issues/608
-
 const (
-	servingDomain          = "knative.dev/serving"
-	internalServingDomain  = "knative.dev/internal/serving"
-	eventingDomain         = "knative.dev/eventing"
-	internalEventingDomain = "knative.dev/internal/eventing"
-	customSubDomain        = "test.domain"
-	testProj               = "test-project"
-	anotherProj            = "another-project"
+	metricsDomain = "knative.dev/project"
```

Comment:
What was the value of having different domains?

Reply:
This was a stack driver thing but we're actually prefixing this to the open census metric names so I think this should keep this around until we resolve: https://github.com/knative/pkg/issues/2174


I think there's some value in repurposing this as a label in order to help disambiguate between components across different Knative projects (ie. serving vs eventing).

ie. in the future we drop the metric prefix and thus require labels to help segment metrics.

So for a golang memory metric - we'd have a component label ie. `controller`  and reuse the domain label `serving`

I'm ok to _drop_ this domain if an alternative exists - ie. if the components namespace is automatically picked up as part of the metric

Reply:
I think the namespace is likely to be picked up as part of the metric; I'm not sure if we have to / want to worry about cases like queue-proxies and sources running in a user's namespace.

I'm inclined to think that namespace + some sort of component/Resource name should be sufficient here, but I might be mistaken.

Reply:
(With the notion that the Resource would indicate whether it was a serving or eventing construct if it was running in a user namespace, and that multi-tenant components would be running in a namespace where the name would give a clue.)

The obvious counter-example here is `mink`, where all the controllers are bundled into a single Pod, but I'm not sure that's a compelling enough case.